### PR TITLE
fix: [CDS-59131]: added connectorRef field for container step

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Screen Shot 2023-05-02 at 1 30 20 PM

### CODE REVIEW

**Review by sonaligoyal-harness — CHANGES_REQUESTED** [review]
To enable allowed values support when this field is made runtime, ConnectorConfigureOptions component need to be added in the code

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/OptionalConfiguration.tsx [hunk 1]
```diff
@@ -382,6 +396,22 @@ export default function OptionalConfiguration(props: {
           style={{ width: 300 }}
         />
       </div>
+      <div>
+        <FormMultiTypeConnectorField
+          label={<Text color={Color.GREY_900}>{getString('connectors.title.harnessImageConnectorRef')}</Text>}
```
Comment (sonaligoyal-harness) [inline src/modules/75-cd/components/PipelineSteps/ContainerStep/OptionalConfiguration.tsx:401]: Nit: label can be added without providing any color and also width for this field looks inconsistent compared to other optional fields

**Comment by sahithikolichala** [comment]
retrigger jest

**Review by sonaligoyal-harness — DISMISSED** [review]

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
worked on increasing coverage, current coverage report is
Screen Shot 2023-05-03 at 4 05 20 PM

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/OptionalConfiguration.tsx [hunk 2]
```diff
@@ -382,6 +396,22 @@ export default function OptionalConfiguration(props: {
...
+        <FormMultiTypeConnectorField
+          label={getString('connectors.title.harnessImageConnectorRef')}
+          type={Connectors.DOCKER}
+          name="spec.infrastructure.spec.harnessImageConnectorRef"
+          placeholder={getString('connectors.placeholder.harnessImageConnectorRef')}
+          accountIdentifier={accountId}
+          projectIdentifier={projectIdentifier}
+          orgIdentifier={orgIdentifier}
+          style={{ marginBottom: 10 }}
+          multiTypeProps={{ allowableTypes: [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME] }}
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ContainerStep/OptionalConfiguration.tsx:409]: Is `expression`  type not needed here? Some fields in the flow has `expression` support some don't.

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger cypress-rest

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Pull request merged**